app/models/user.rb

```diff
@@ -450,6 +450,11 @@ def enough_credits?(num_credits_needed)
     credits.unspent.size >= num_credits_needed
   end
 
+  def receives_follower_notifications?
```

Comment:
Are there other types of notifications, i.e. should the name of this method contain `email`?

Reply:
I'm not totally sure about this method, because it makes `User` a special followable case (the other followables don't have this method) and keeps readability the same. Though the job is only related to `User` object, followables in the code base can be tags, organizations and users. 

I don't know how to explain it correctly but we go from:

```ruby
if followable && followable.email && followable.email_follower_notifications
```

to

```ruby
if followable && followable.receives_email_follower_notifications

...

def receives_email_follower_notifications
  email && email_follower_notifications
end
```

I wonder if it wouldn't be better not to refactor the worker at this point :)

Reply:
Thanks for the review!

I wasn't thinking about readability here, but I was concerned that the logic, whether a user wants to receive notifications or not, belongs to the user, not the worker, thus the worker has to ask the user object.

Indeed, it makes a User a special type of follower, which is kind of an interface here. I wanted to put it into a shared follower module, but none of the other followable models have `email_follower_notifications` implemented.

Addressing all other issues now. Do you prefer moving this logic to some `models/shared/follower.rb` module and add `self.class.name == "User"` to have a more strict separation of logic? If not, I'll put back the original guard clause 🙂

Reply:
Thanks for the explanation, let's keep it like you did!

Reply:
Please add the word "email" to the method name and we're done! :)